File: Angular+.NET+MSSQL/PatientApp/PatientAPI/PatientAPI/Models/PatientModel.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace PatientAPI.Models
{
    public class PatientModel
    {
        [Required(ErrorMessage = "First Name is required")]
        [MinLength(2, ErrorMessage = "First Name must be at least 2 characters long")]
        public string FirstName { get; set; }

        [Required(ErrorMessage = "Last Name is required")]
        [MinLength(2, ErrorMessage = "Last Name must be at least 2 characters long")]
        public string LastName { get; set; }

        [Required(ErrorMessage = "Gender is required")]
        [JsonConverter(typeof(JsonStringEnumConverter))]
        public GenderType? Gender { get; set; }

        [Required(ErrorMessage = "Date of Birth is required")]
        public DateTime  DOB { get; set; }

        [Required(ErrorMessage = "Phone Number is required")]
        [RegularExpression(@"^\d{10}$", ErrorMessage = "Phone Number must be 10 digits long")]
        public string PhoneNo { get; set; }

        [Required(ErrorMessage = "Address is required")]
        [MinLength(5, ErrorMessage = "Address must be at least 5 characters long")]
        [MaxLength(100, ErrorMessage = "Address cannot exceed 100 characters")]
        public string Address { get; set; }

        [Required(ErrorMessage = "File can't be empty")]
        public IFormFile File { get; set; }
    }
}

File: Angular+.NET+MSSQL/PatientApp/PatientUI/src/app/models/patient.model.ts
import { Gender } from "./Gender";
export interface Patient {
    id?: number; // made the id field nullable
    firstname: string;
    lastname: string;
    gender: Gender; // added gender field with optional type
    dob?: Date; // added date of birth field with optional type
    phoneNo: string;
    address?: string; // added address field with optional type
}
// {
//     "firstName": "string",
//     "lastName": "string",
//     "gender": "string",
//     "dateOfBirth": "2024-05-24T05:43:47.271Z",
//     "phoneNumber": "5227653212",
//     "address": "string"
//   }